test/TaskQueue_test.js

```diff
@@ -0,0 +1,87 @@
+const assert = require('assert');
+const debug = require('debug')('index:test:api_test');
+const helper = require('./helper');
+const slugid = require('slugid');
+const taskcluster = require('taskcluster-client');
+const request = require('superagent');
+const assume = require('assume');
+const libUrls = require('taskcluster-lib-urls');
+
+helper.secrets.mockSuite('TaskQueue_test.js', ['taskcluster'], function(mock, skipping) {
+  helper.withFakeQueue(mock, skipping);
+  const makeTask = function() {
+    return {
+      provisionerId: 'garbage-hybrid1999',
+      workerType: 'fail',
+      scopes: [],
+      retries: 3,
+      created: (new Date()).toJSON(),
+      deadline: (new Date()).toJSON(),
+      expires: taskcluster.fromNow('1 day'),
+      payload: {nigga:1},
+      tags: {
+        objective: 'Test task indexing',
+      },
+    };
+  };
+
+  test('Run task and test indexing', async function() {
+    const tq = await helper.load('taskqueue');
+    helper.claimableWork.push({
+      tasks: {
+        status: {
+          taskId:0,
+        },
+        runId:0,
+        workerGroup:'garbage-hybrid1999',
+        workerId:'succeed',
+        task: {
+          provisionerId: 'garbage-hybrid1999',
+          workerType:'succeed',
+          payload:{},
+        },
+      },
+    });
+    await tq.claimTasks();
+    assert.deepEqual(helper.taskResolutions[0], {completed: true});
```

Comment:
After the task is resolved as fail or succeed,  the result is stored in the taskResolution array. I checked if the output is corrrect in Line 46. The outputs are stored in taskResolution

Reply:
`taskResolutions` basically contains the side-effects of `claimTasks`.  One way we could make that more readable is to add a helper function `helper.assertTaskResolved(taskId, resolution)`.

Quick note while I'm here; taskIds are strings, not integers, so it's a bit confusing to see `0` as a taskId here.  The `slugid` library will generate taskIds for you, so this might be best as:

```javascrpit
  const taskId = slugid.nice();
  helper.claimableWork.push({
    ...
    status: {taskId},
    ...
  });

  helper.assertTaskResolved(taskId, {completed: true});
```